shell/platform/darwin/ios/framework/Source/TextInputSemanticsObject.mm

```diff
@@ -41,7 +41,9 @@ - (NSString*)textInRange:(UITextRange*)range {
   NSAssert([range isKindOfClass:[FlutterTextRange class]],
            @"Expected a FlutterTextRange for range (got %@).", [range class]);
   NSRange textRange = ((FlutterTextRange*)range).range;
-  NSAssert(textRange.location != NSNotFound, @"Expected a valid text range.");
+  if (textRange.location == NSNotFound) {
```

Comment:
Could you add a test to verify that `textInRange:` does not crash if the input range's location is `NSNotFound`?